### PR TITLE
Remove download test json

### PR DESCRIPTION
**Pull request by gal-forer**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-hq.paloaltonetworks.local/browse/CIAC-3856
## Description
replace json file for the. secretes
## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/scripts/download_demisto_conf.sh [hunk 1]
```diff
@@ -28,7 +28,9 @@ if [ "$?" != "0" ]; then
 fi
 cp -r ./content-test-conf/awsinstancetool ./Tests/scripts/awsinstancetool
 cp -r ./content-test-conf/demisto.lic $DEMISTO_LIC_PATH
-cp -r ./content-test-conf/conf.json $SECRET_CONF_PATH
+echo "Starting the secret file creation"
+echo "^^^^^^^^^^^^^^^^^^: " $SECRET_CONF_PATH
+python3 ./Tests/scripts/add_secret_file_to_build.py $SECRET_CONF_PATH
```
Comment (esharf) [inline Tests/scripts/download_demisto_conf.sh:33]: Let’s move it to different step (it shouldn’t be a part of this script anymore)

**Review by esharf — DISMISSED** [review]

File: Tests/scripts/utils/GoogleSecretManagerModule.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        for secret in self.client.list_secrets(request={"parent": parent}):
+            secret.name = str(secret.name).split('/')[-1]
+            if name_filter and name_filter not in secret.name:
+                continue
+            if with_secret:
+                try:
+                    secret_value = self.get_secret(project_id, secret.name)
+                    secrets.append(secret_value)
+                except google.api_core.exceptions.NotFound:
+                    pass
```
[the remark this answers is not in the document]
Comment (gal-forer) [reply Tests/scripts/utils/GoogleSecretManagerModule.py:29]: Yes I didn't know how to

File: Tests/scripts/add_secret_file_to_build.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import argparse
+import json
+
+from Tests.scripts.utils.GoogleSecretManagerModule import GoogleSecreteManagerModule
+
+
+def run(options):
+    json_path_file = options.service_account
+    print(f'******************conf file location: {json_path_file}')
```
Comment (tkatzir) [inline Tests/scripts/add_secret_file_to_build.py:9]: Also, remove all the *'s before merging.
```suggestion
    print(f'****************** conf file location: {json_path_file}')
```

File: Tests/scripts/add_secret_file_to_build.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    project_id = options.gsm_project_id
+    secrets = secret_conf.list_secrets(project_id, with_secret=True)
+    secret_file = {
+        "username": options.user,
+        "userPassword": options.password,
+        "integrations": secrets
+    }
+
+    print(f'secrets from API: {secret_file[0:10]}')
+    print(f'json_path_file: {json_path_file}')
```
Comment (tkatzir) [inline Tests/scripts/add_secret_file_to_build.py:23]:
```suggestion
    print(f'{json_path_file=}')
```

File: Tests/scripts/add_secret_file_to_build.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    with open(json_path_file, 'w') as secrets_out_file:
+        secrets_out_file.write(json.dumps(secret_file))
+    print(json_path_file)
+
+
+def options_handler(args=None):
+    parser = argparse.ArgumentParser(description='Utility for Importing secrets from Google Secret Manager.')
+    parser.add_argument('-pid', '--gsm_project_id', help='The project id for the GSM.')
```
Comment (tkatzir) [inline Tests/scripts/add_secret_file_to_build.py:31]: Maybe rename it to `gpid`? `pid` is a widely used standard for `process id`.

File: Tests/scripts/add_secret_file_to_build.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    parser.add_argument('-u', '--user', help='Path to secret xsiam server metadata file.')
+    parser.add_argument('-p', '--password', help='Path to file with XSIAM Servers api keys.')
```
Comment (tkatzir) [inline Tests/scripts/add_secret_file_to_build.py:33]: Is `password` the best name for this parameter?...

File: Tests/scripts/add_secret_file_to_build.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    # disable-secrets-detection-start
+    parser.add_argument('-sa', '--service_account',
+                        help=("Path to gcloud service account, is for circleCI usage. "
+                              "For local development use your personal account and "
+                              "authenticate using Google Cloud SDK by running: "
+                              "`gcloud auth application-default login` and leave this parameter blank. "
+                              "For more information go to: "
```
Comment (tkatzir) [inline Tests/scripts/add_secret_file_to_build.py:40]:
```suggestion
                              "For more information see: "
```

File: Tests/scripts/add_secret_file_to_build.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+                              "https://googleapis.dev/python/google-api-core/latest/auth.html"),
+                        required=False)
+    # disable-secrets-detection-end
+    options = parser.parse_args(args)
+
+    return options
+
+
+def get_secret_service_account_config(json_path_file: str) -> str:
+    print('##################################')
```
Comment (tkatzir) [inline Tests/scripts/add_secret_file_to_build.py:50]: Remove before merging.

File: Tests/scripts/download_demisto_conf.sh [hunk 9]
```diff
@@ -28,7 +28,10 @@ if [ "$?" != "0" ]; then
 fi
 cp -r ./content-test-conf/awsinstancetool ./Tests/scripts/awsinstancetool
 cp -r ./content-test-conf/demisto.lic $DEMISTO_LIC_PATH
-cp -r ./content-test-conf/conf.json $SECRET_CONF_PATH
+echo "Starting the secret file creation"
+echo "^^^^^^^^^^^^^^^^^^: " $SECRET_CONF_PATH
```
Comment (tkatzir) [inline Tests/scripts/download_demisto_conf.sh:32]: Make it cleaner before merging.

File: Tests/scripts/utils/GoogleSecretManagerModule.py [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+
+    @staticmethod
+    def init_secret_manager_client(service_account: str) -> secretmanager.SecretManagerServiceClient:
+        try:
+            service_account = service_account.replace('\r', '').replace('\n', '')
+            cur_directory_path = os.getcwd()
+            credentials_file_name = f'{datetime.now().strftime("%m-%d-%Y,%H:%M:%S:%f")}.json'
+            credentials_file_path = os.path.join(cur_directory_path, credentials_file_name)
+            json_object = json.loads(service_account)
+            f = open(credentials_file_path, "w")
```
Comment (tkatzir) [inline Tests/scripts/utils/GoogleSecretManagerModule.py:44]: Change to `with ...`

File: Tests/scripts/utils/GoogleSecretManagerModule.py [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+            f.write(json.dumps(json_object))
+            f.close()
+            client = secretmanager.SecretManagerServiceClient.from_service_account_json(credentials_file_path)# type: ignore # noqa
+            return client
+        finally:
+            try:
+                os.remove(credentials_file_path)
```
Comment (tkatzir) [inline Tests/scripts/utils/GoogleSecretManagerModule.py:51]: Good to put it in the `finally` block! :-)

**Review by tkatzir — COMMENTED** [review]
Good progress!
Some comments.

File: Tests/scripts/add_secret_file_to_build.py [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+
+def options_handler(args=None):
+    parser = argparse.ArgumentParser(description='Utility for Importing secrets from Google Secret Manager.')
+    parser.add_argument('-gpid', '--gsm_project_id', help='The project id for the GSM.')
+    parser.add_argument('-u', '--user', help='the user for Demisto.')
+    parser.add_argument('-p', '--password', help='The password for Demisto.')
+    parser.add_argument('-sf', '--json_path_file', help='Path to the secret json file.')
+    # disable-secrets-detection-start
+    parser.add_argument('-sa', '--service_account',
+                        help=("Path to gcloud service account, is for circleCI usage. "
```
Comment (tkatzir) [inline Tests/scripts/add_secret_file_to_build.py:40]:
```suggestion
                        help=("Path to gcloud service account, for circleCI usage. "
```

File: Tests/scripts/utils/GoogleSecretManagerModule.py [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+            logging.error(f'Secret json is malformed for: {secret_id}, got error: {e}')
+
+    def list_secrets(self, project_id: str, name_filter: list = [], with_secret=False) -> list:
+        secrets = []
+        parent = f"projects/{project_id}"
+        print(f'parent: {parent}')
+
+        for secret in self.client.list_secrets(request={"parent": parent}):
+            secret.name = str(secret.name).split('/')[-1]
+            print(f'______________________{secret.name}______________________')
```
Comment (tkatzir) [inline Tests/scripts/utils/GoogleSecretManagerModule.py:26]:
```suggestion
            print(f'______________________ {secret.name} ______________________')
```

**Review by tkatzir — COMMENTED** [review]
Good progress!
Some comments.

File: Tests/scripts/utils/GoogleSecretManagerModule.py [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+    def __init__(self, service_account_file: str):
+        self.client = self.init_secret_manager_client(service_account_file)
+
+    def get_secret(self, project_id: str, secret_id: str, version_id: str = 'latest') -> dict:
+        name = f"projects/{project_id}/secrets/{secret_id}/versions/{version_id}"
+        response = self.client.access_secret_version(request={"name": name})
+        try:
+            return json5.loads(response.payload.data.decode("UTF-8"))
+        except Exception as e:
+            logging.error(f'Secret json is malformed for: {secret_id}, got error: {e}')
```
[the remark this answers is not in the document]
Comment (gal-forer) [reply Tests/scripts/utils/GoogleSecretManagerModule.py:17]: I've added the version to the log

File: .gitlab/ci/bucket-upload.yml [hunk 15]
```diff
@@ -169,6 +174,11 @@ install-packs-in-server-master:
     - EXIT_CODE=0
     - section_start "Download configuration"
     - ./Tests/scripts/download_demisto_conf.sh  >> $ARTIFACTS_FOLDER/logs/installations.log
+    - SECRET_CONF_PATH=$(cat secret_conf_path)
+    - echo "T_T" >> $ARTIFACTS_FOLDER/testKKK.txt
+    - ls >> $ARTIFACTS_FOLDER/lsKKK.txt
+    - echo "b2"
```
Comment (DeanArbel) [inline .gitlab/ci/bucket-upload.yml:180]: Please remove
```suggestion
```

File: .gitlab/ci/bucket-upload.yml [hunk 16]
```diff
@@ -106,6 +106,11 @@ prepare-testing-bucket-xpanse-upload-flow:
     - EXIT_CODE=0
     - section_start "Download configuration"
     - ./Tests/scripts/download_demisto_conf.sh  >> $ARTIFACTS_FOLDER/logs/installations.log
+    - SECRET_CONF_PATH=$(cat secret_conf_path)
+    - echo "T_T" >> $ARTIFACTS_FOLDER/testKKK.txt
+    - ls >> $ARTIFACTS_FOLDER/lsKKK.txt
+    - echo "b1"
```
Comment (DeanArbel) [inline .gitlab/ci/bucket-upload.yml:112]: Please remove
```suggestion
```

File: .gitlab/ci/global.yml [hunk 17]
```diff
@@ -55,6 +55,11 @@
 .download-demisto-conf:
   - section_start "Download content-test-conf" --collapsed
   - ./Tests/scripts/download_demisto_conf.sh  >> $ARTIFACTS_FOLDER/logs/download_demisto_conf.log
+  - SECRET_CONF_PATH=$(cat secret_conf_path)
+  - echo "T_T" >> $ARTIFACTS_FOLDER/testKKK.txt
+  - ls >> $ARTIFACTS_FOLDER/lsKKK.txt
+  - echo "global"
```
Comment (DeanArbel) [inline .gitlab/ci/global.yml:61]: Please remove
```suggestion
```

File: Packs/Bitbucket/Integrations/Bitbucket/Bitbucket.py [hunk 18]
```diff
@@ -1410,7 +1410,7 @@ def pull_request_comment_create_command(client: Client, args: Dict) -> CommandRe
     outputs = response
     if partial_response:
         outputs = comment_partial_result(response, 'pr')
-    return CommandResults(readable_output='The comment was created successfully',
+    return CommandResults(readable_output='The comment was created successfullyTTTTTTTTT',
```
Comment (DeanArbel) [inline Packs/Bitbucket/Integrations/Bitbucket/Bitbucket.py:1413]: Please clean this
```suggestion
    return CommandResults(readable_output='The comment was created successfully',
```

File: Packs/FeedURLhaus/Integrations/FeedURLhaus/FeedURLhaus.py [hunk 19]
```diff
@@ -31,7 +31,7 @@ def main():  # pragma: no cover
         params['url'] = chosen_urls
         feed_main('URLhaus Feed', params, 'urlhaus-')
     except Exception as e:
-        return_error(f'Failed to execute {demisto.command()} command. Error: {str(e)}')
+        return_error(f'Failed toKKK execute {demisto.command()} command. Error: {str(e)}')
```
Comment (DeanArbel) [inline Packs/FeedURLhaus/Integrations/FeedURLhaus/FeedURLhaus.py:34]: Please clean
```suggestion
        return_error(f'Failed to execute {demisto.command()} command. Error: {str(e)}')
```

File: Tests/scripts/add_secrets_file_to_build.py [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+    logging.info(f'saved the json file to: {options.json_path_file}')
+
+
+def options_handler(args=None):
+    parser = argparse.ArgumentParser(description='Utility for Importing secrets from Google Secret Manager.')
+    parser.add_argument('-gpid', '--gsm_project_id', help='The project id for the GSM.')
+    parser.add_argument('-u', '--user', help='the user for Demisto.')
+    parser.add_argument('-p', '--password', help='The password for Demisto.')
+    parser.add_argument('-sf', '--json_path_file', help='Path to the secret json file.')
+    parser.add_argument('-a', '--ar', help='Test.')
```
Comment (DeanArbel) [inline Tests/scripts/add_secrets_file_to_build.py:46]: What does this argument do?

**Review by DeanArbel — CHANGES_REQUESTED** [review]

File: Tests/scripts/add_secrets_file_to_build.py (hunk 20 above)
Comment (gal-forer) [reply Tests/scripts/add_secrets_file_to_build.py:46]: I removed it

**Review by DeanArbel — APPROVED** [review]
Nicely done

**Pull request merged**